src/components/core/main-nav.tsx

```diff
@@ -7,10 +7,18 @@ import { usePathname } from "next/navigation"
 import { siteConfig } from "@/config/site"
 import { cn } from "@/lib/utils"
 import { Icons } from "@/components/core/icons"
+import {
+  selectUser,
+  useSelector,
+} from '@/lib/redux'
+import { User } from "@/lib/redux/slices/authSlice/models";
 
 export function MainNav({ isPublic }: { isPublic: boolean }) {
   const pathname = usePathname()
 
+  const user = useSelector<User>(selectUser)
+  const dashboardHref = "/pitwall/dashboard/" + user.pitBoxSessionId
```

Comment:
I think this is only safe if it's not public nav, otherwise `selectUser` will throw an error. I wanted to keep `selectUser` from being nullable, to make it easier to work with in places where we _know_ there is a user, but maybe being nullable makes the most sense.

Reply:
I have pushed now a change to no longer throw the error. It's not the optimal solution, but I think we can live with that for now.